### PR TITLE
ZKR-3672-Benchmark

### PR DESCRIPTION
**Pull request by FatemehHeidari**
New benchmark scenarios are added for a new circuit with multiple identical lookups. 

Some functionality for comparing the benchmark results was added to compare the performance between version one, and different methods of handling lookup in version 2.

### CODE REVIEW

File: korrekt/src/sample_circuits/pse_v1/lookup_circuits/multiple_matched_lookups.rs [hunk 1]
```diff
@@ -0,0 +1,348 @@
...
+                        self.config.s_range.enable(&mut region, row)?;
+                        self.config.s_range_1.enable(&mut region, row)?;
+                        region.assign_advice(
+                            || "advice",
+                            self.config.advice[2],
+                            row,
+                            || {
+                                b_cell.value().and_then(|a| {
+                                    c_cell.value().map(|b| {
+                                        let a_val = u64::from_str_radix(format!("{:?}",a).strip_prefix("0x").unwrap(), 16).unwrap();//a.get_lower_32() as u64;
```
Comment (jgorzny) [inline korrekt/src/sample_circuits/pse_v1/lookup_circuits/multiple_matched_lookups.rs:290]:
```suggestion
                                        let a_val = u64::from_str_radix(format!("{:?}",a).strip_prefix("0x").unwrap(), 16).unwrap();
```

File: korrekt/src/sample_circuits/pse_v1/lookup_circuits/multiple_matched_lookups.rs [hunk 2]
```diff
@@ -0,0 +1,348 @@
...
+                                        let b_val = u64::from_str_radix(format!("{:?}",b).strip_prefix("0x").unwrap(), 16).unwrap();//b.get_lower_32() as u64;
```
Comment (jgorzny) [inline korrekt/src/sample_circuits/pse_v1/lookup_circuits/multiple_matched_lookups.rs:291]:
```suggestion
                                        let b_val = u64::from_str_radix(format!("{:?}",b).strip_prefix("0x").unwrap(), 16).unwrap();
```

File: korrekt/src/io/analyzer_io.rs [hunk 3]
```diff
@@ -114,11 +115,11 @@ pub fn retrieve_user_input_for_analyzer_type() -> Result<AnalyzerType> {
...
-    println!("Choose the mode of analysis for your circuit.");
-    println!("1. Unused Gates");
-    println!("2. Unused Columns");
-    println!("3. Unconstrained Cells");
-    println!("4. Underconstrained Circuit");
+    info!("Choose the mode of analysis for your circuit.");
+    info!("1. Unused Gates");
+    info!("2. Unused Columns");
+    info!("3. Unconstrained Cells");
+    info!("4. Underconstrained Circuit");
```
Comment (jgorzny) [inline korrekt/src/io/analyzer_io.rs:122]: Should these actually be `info` over `println`? I am not sure -- but I would have thought options go to stdout by default. Is there a rust convention that should be followed? Not a scientific answer, but https://users.rust-lang.org/t/how-to-get-user-input/5176/11 suggests `println`.
Or was there another reason to implement it this way?

Comment (FatemehHeidari) [reply korrekt/src/io/analyzer_io.rs:122]: Yes, that's correct. For running benchmarks I disabled all the `println`.

However, upon further review, the `retrieve_user_input_for_analyzer_type()` function could be removed, as we now receive user inputs through CLI options.

**Review by jgorzny — APPROVED** [review]

**Comment by FatemehHeidari** [comment]
ZKR-3672-Benchmark

**Pull request merged**